compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1498,6 +1498,7 @@ public boolean isInherentlyImmutableType(BType type) {
             case TypeTags.INVOKABLE:
             case TypeTags.TYPEDESC:
             case TypeTags.HANDLE:
+            case TypeTags.REGEXP:    
```

Comment:
Need to update the runtime type checker also. Please check how we can add tests for that too.

Reply:
Updated in [678ba28](https://github.com/ballerina-platform/ballerina-lang/pull/41843/commits/678ba2815e15240bb6accfae597acec00e0e5455)